types.go

```diff
@@ -14,12 +16,64 @@ type Source struct {
 	Repository string `json:"repository"`
 	RawTag     Tag    `json:"tag,omitempty"`
 
-	Username string `json:"username,omitempty"`
-	Password string `json:"password,omitempty"`
+	Username     string       `json:"username,omitempty"`
+	Password     string       `json:"password,omitempty"`
+	ContentTrust ContentTrust `json:"content_trust,omitempty"`
 
 	Debug bool `json:"debug,omitempty"`
 }
 
+type ContentTrust struct {
+	Enable               bool   `json:"enable"`
+	Server               string `json:"server"`
+	RepositoryKeyID      string `json:"repository_key_id"`
+	RepositoryKey        string `json:"repository_key"`
+	RepositoryPassphrase string `json:"repository_passphrase"`
+	TLSKey               string `json:"tls_key"`
+	TLSCert              string `json:"tls_cert"`
+}
+
+/* Create notary config directory with following structure
+├── gcr-config.json
+└── trust
+	└── private
+		└── <private-key-id>.key
+└── tls
+	└── <notary-host>
+		├── client.cert
+		└── client.key
+*/
+func (ct *ContentTrust) PrepareConfigDir(src string) (string, error) {
```

Comment:
This also seems a little odd. Why does it have to be written to disk? Is this for integrating with an external tool, or could it be changed to just pass a config argument to your library? :thinking:

Reply:
Yes, many API from the official `github.com/theupdateframework/notary` rely on a base directory to store where all the trust files. For example, https://github.com/theupdateframework/notary/blob/28e8e6d5c2221984e9acd1f2bab017cfdcdf106c/client/client.go#L60.